time t
int32 x
int32 y
bool resp

